File: src/cpp/FirstNotRepeatingChar.cpp
class Solution {
public:
    int FirstNotRepeatingChar(string str) {
        if (str.length() == 0) {
            return -1;
        }
        int hash[256] = {0};
        int i = 0;
        while (str[i] != '\0') {
            hash[str[i]]++;
            i++;
        }
        i = 0;
        while (str[i] != '\0') {
            if (1 == hash[str[i]]) {
                return i;
            }
            i++;
        }
        return -1;
    }
};

File: src/java/FirstNotRepeatingChar.java
public class Solution {
    public int FirstNotRepeatingChar(String str) {
        char[] c = str.toCharArray();
        int[] a = new int[256];
        for (char d : c) {
            a[(int) d]++;
        }

        for (int i = 0; i < c.length; i++) {
            if (a[(int) c[i]] == 1) {
                return i;
            }
        }

        return -1;
    }
}
